src/integrationTest/java/org/opensearch/security/SnapshotSteps.java

```diff
@@ -0,0 +1,87 @@
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+package org.opensearch.security;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.awaitility.Awaitility;
+
+import org.opensearch.action.admin.cluster.repositories.delete.DeleteRepositoryRequest;
+import org.opensearch.action.admin.cluster.repositories.put.PutRepositoryRequest;
+import org.opensearch.action.admin.cluster.snapshots.create.CreateSnapshotRequest;
+import org.opensearch.action.admin.cluster.snapshots.create.CreateSnapshotResponse;
+import org.opensearch.action.admin.cluster.snapshots.delete.DeleteSnapshotRequest;
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest;
+import org.opensearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.opensearch.action.admin.cluster.snapshots.restore.RestoreSnapshotRequest;
+import org.opensearch.action.admin.cluster.snapshots.restore.RestoreSnapshotResponse;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.SnapshotClient;
+import org.opensearch.snapshots.SnapshotInfo;
+import org.opensearch.snapshots.SnapshotState;
+
+import static java.util.Objects.requireNonNull;
+import static org.opensearch.client.RequestOptions.DEFAULT;
+
+class SnapshotSteps {
+
+	private final SnapshotClient snapshotClient;
+
+	public SnapshotSteps(RestHighLevelClient restHighLevelClient) {
+		this.snapshotClient = requireNonNull(restHighLevelClient, "Rest high level client is required.").snapshot();
+	}
+
+	//CS-SUPPRESS-ALL: It is not possible to use phrase "cluster manager" instead of master here
+	public org.opensearch.action.support.master.AcknowledgedResponse createSnapshotRepository(String repositoryName, String snapshotDirPath, String type)
+	//CS-ENFORCE-ALL
+		throws IOException {
+		PutRepositoryRequest createRepositoryRequest = new PutRepositoryRequest().name(repositoryName).type(type)
+			.settings(Map.of("location", snapshotDirPath));
+		return snapshotClient.createRepository(createRepositoryRequest, DEFAULT);
+	}
+
+	public CreateSnapshotResponse createSnapshot(String repositoryName, String snapshotName, String...indices) throws IOException {
+		CreateSnapshotRequest createSnapshotRequest = new CreateSnapshotRequest(repositoryName, snapshotName)
+			.indices(indices);
+		return snapshotClient.create(createSnapshotRequest, DEFAULT);
+	}
+
+	public void waitForSnapshotCreation(String repositoryName, String snapshotName) {
+		GetSnapshotsRequest getSnapshotsRequest = new GetSnapshotsRequest(repositoryName, new String[] { snapshotName });
+		Awaitility.await().until(() -> {
+			GetSnapshotsResponse snapshotsResponse = snapshotClient.get(getSnapshotsRequest, DEFAULT);
+			SnapshotInfo snapshotInfo = snapshotsResponse.getSnapshots().get(0);
+			return SnapshotState.SUCCESS.equals(snapshotInfo.state());
+		});
+	}
+
+	//CS-SUPPRESS-ALL: It is not possible to use phrase "cluster manager" instead of master here
```

Comment:
Please do not use SUPRESS-ALL, instead use something like `// CS-SUPPRESS-SINGLE: RegexpSingleline https://github.com/opensearch-project/OpenSearch/issues/3663` that will only suppress a single kind of violation

Reply:
done